backend/main/main-canister.mo

```diff
@@ -457,29 +465,57 @@ actor {
 	stable var packagePublicationsStable: [(PackageId, PackagePublication)] = [];
 	stable var packageVersionsStable: [(PackageName, [Ver])] = [];
 	stable var packageOwnersStable: [(PackageName, Principal)] = [];
-	stable var packageConfigsStable: [(PackageId, PackageConfig)] = [];
-	stable var highestConfigsStable: [(PackageName, PackageConfig)] = [];
+	stable var packageConfigsStableV2: [(PackageId, PackageConfigV2)] = [];
+	stable var highestConfigsStableV2: [(PackageName, PackageConfigV2)] = [];
 
 	stable var fileIdsByPackageStable: [(PackageId, [FileId])] = [];
 
 	stable var downloadLogStable: DownloadLog.Stable = null;
 	stable var storageManagerStable: StorageManager.Stable = null;
 
+	// deprecated stable collections
+	stable var packageConfigsStable: [(PackageId, PackageConfig)] = [];
+	stable var highestConfigsStable: [(PackageName, PackageConfig)] = [];
+
+	func migrateConfigs(array: [(PackageName, PackageConfig)]): [(PackageName, PackageConfigV2)]{
+		Array.map<(PackageName, PackageConfig), (PackageName, PackageConfigV2)>(
+			array,
+			func((name, config)){
+				let newConfig = {
+					config with dependencies = Array.map(
+						config.dependencies,
+						func(dep: Dependency): DependencyV2 {
+							{ dep with repo = ""}
```

Comment:
new syntax, nice)

Reply:
yh, it's pretty cool. It was shipped in the `0.7.0` version of the motoko compiler.